### PR TITLE
Remove duplicate_dequantize_node and remove_extra_dequantize

### PR DESCRIPTION
**Pull request by danrecoskie**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #83611
* #83459

Summary: removed duplicate_dequantize_node and remove_extra_dequantize

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
python test/test_quantization.py TestQuantizeFxModels

Reviewers: jerryzh168

Subscribers:

Tasks:

Tags:

Differential Revision: [D38841052](https://our.internmc.facebook.com/intern/diff/D38841052)

### CODE REVIEW

File: torch/ao/quantization/fx/_lower_to_native_backend.py [hunk 1]
```diff
@@ -952,8 +934,8 @@ def _lower_to_native_backend(
     _lower_quantized_binary_op(model, qconfig_map)
     _lower_getattr_tensor_metadta_op(model)
     special_pattern_replacement(model)
-    model = fold_weight(model, node_name_to_scope)
     model.graph.eliminate_dead_code()
+    model = fold_weight(model, node_name_to_scope)
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/_lower_to_native_backend.py:938]: maybe run `eliminate_dead_code` again after this, just to be safe

**Review by jerryzh168 — APPROVED** [review]
looks great, thanks!

**Comment by jerryzh168** [comment]
can you add more details to the summary describing what we did? or you can just link the issue

**Comment by danrecoskie** [comment]
@danrecoskie has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38841052).

File: torch/ao/quantization/fx/_lower_to_native_backend.py [hunk 2]
```diff
@@ -952,6 +933,7 @@ def _lower_to_native_backend(
     _lower_quantized_binary_op(model, qconfig_map)
     _lower_getattr_tensor_metadta_op(model)
     special_pattern_replacement(model)
+    model.graph.eliminate_dead_code()
     model = fold_weight(model, node_name_to_scope)
     model.graph.eliminate_dead_code()
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/_lower_to_native_backend.py:938]: looks like `model.graph.eliminate_dead_code()` is missing after 937 here?

Comment (jerryzh168) [reply torch/ao/quantization/fx/_lower_to_native_backend.py:938]: this looks good, sorry I misread this code before

**Pull request closed without merging**